webapi/Extensions/SemanticKernelExtensions.cs

```diff
@@ -117,15 +116,22 @@ public static IKernel RegisterChatSkill(this IKernel kernel, IServiceProvider sp
     }
 
     /// <summary>
-    /// Propagate exception from within semantic function
+    /// Invokes an asynchronous callback function and tags any exception that occurs with function name.
     /// </summary>
-    public static void ThrowIfFailed(this SKContext context)
+    /// <typeparam name="T">The type of the result returned by the callback function.</typeparam>
+    /// <param name="callback">The asynchronous callback function to invoke.</param>
+    /// <param name="functionName">The name of the function that calls this method, for logging purposes.</param>
+    /// <returns>A task that represents the asynchronous operation and contains the result of the callback function.</returns>
+    public static async Task<T> SafeInvokeAsync<T>(Func<Task<T>> callback, string functionName)
     {
-        if (context.ErrorOccurred)
+        try
         {
-            var logger = context.LoggerFactory.CreateLogger(nameof(SKContext));
-            logger.LogError(context.LastException, "{0}", context.LastException?.Message);
-            throw context.LastException!;
```

Comment:
Removed logging here because it's already logged upstream (I think when TrackSkillFunction is passed a failure)
![image](https://github.com/microsoft/chat-copilot/assets/125500434/4f261f1d-81bf-4768-8db0-18d2d4f666f4)